examples/cars/gen/http/cars/client/encode_decode.go

```diff
@@ -0,0 +1,195 @@
+// Code generated by goa v2.0.0-wip, DO NOT EDIT.
+//
+// cars HTTP client encoders and decoders
+//
+// Command:
+// $ goa gen goa.design/goa/examples/cars/design -o
+// $(GOPATH)/src/goa.design/goa/examples/cars
+
+package client
+
+import (
+	"bytes"
+	"context"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+
+	carssvc "goa.design/goa/examples/cars/gen/cars"
+	goahttp "goa.design/goa/http"
+)
+
+// BuildLoginRequest instantiates a HTTP request object with method and path
+// set to call the "cars" service "login" endpoint
+func (c *Client) BuildLoginRequest(ctx context.Context, v interface{}) (*http.Request, error) {
+	u := &url.URL{Scheme: c.scheme, Host: c.host, Path: LoginCarsPath()}
+	req, err := http.NewRequest("POST", u.String(), nil)
+	if err != nil {
+		return nil, goahttp.ErrInvalidURL("cars", "login", u.String(), err)
+	}
+	if ctx != nil {
+		req = req.WithContext(ctx)
+	}
+
+	return req, nil
+}
+
+// EncodeLoginRequest returns an encoder for requests sent to the cars login
+// server.
+func EncodeLoginRequest(encoder func(*http.Request) goahttp.Encoder) func(*http.Request, interface{}) error {
+	return func(req *http.Request, v interface{}) error {
+		p, ok := v.(*carssvc.LoginPayload)
+		if !ok {
+			return goahttp.ErrInvalidType("cars", "login", "*carssvc.LoginPayload", v)
+		}
+		req.SetBasicAuth(p.User, p.Password)
+		return nil
+	}
+}
+
+// DecodeLoginResponse returns a decoder for responses returned by the cars
+// login endpoint. restoreBody controls whether the response body should be
+// restored after having been read.
+// DecodeLoginResponse may return the following errors:
+//	- "unauthorized" (type carssvc.Unauthorized): http.StatusUnauthorized
+//	- error: internal error
+func DecodeLoginResponse(decoder func(*http.Response) goahttp.Decoder, restoreBody bool) func(*http.Response) (interface{}, error) {
+	return func(resp *http.Response) (interface{}, error) {
+		if restoreBody {
+			b, err := ioutil.ReadAll(resp.Body)
+			if err != nil {
+				return nil, err
+			}
+			resp.Body = ioutil.NopCloser(bytes.NewBuffer(b))
+			defer func() {
+				resp.Body = ioutil.NopCloser(bytes.NewBuffer(b))
+			}()
+		} else {
+			defer resp.Body.Close()
```

Comment:
So I'm getting a segfault from this line under the following circumstances:
Service A has a websocket endpoint. Service B subscribes to that endpoint. Service A restarts -- Service B then segfaults at this code. B automatically restarts then is fine. If you trace back when this code is calls when err != nil from a websocket dial -- resp is nil in a number of those cases it seems.

Reply:
thanks, seems like we should wrap the close with a `nil` check.